Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskSet.cs

```diff
@@ -61,37 +66,56 @@ protected override void DisposeSelf()
             {
                 CancelRequestsContexts.Dispose();
             }
-            m_MigratableEntityPersistentDataByType.DisposeAllValuesAndClear();
 
             base.DisposeSelf();
         }
 
         public void AddResolvableDataStreamsTo(Type type, List<AbstractDataStream> dataStreams)
         {
-            if (!m_PublicDataStreamsByType.TryGetValue(type, out AbstractDataStream dataStream))
+            if (m_Debug_TypeCount.TryGetValue(type, out int count) && count > 1)
+            {
+                throw new InvalidOperationException($"Trying to get resolvable data streams for the type {type.GetReadableName()} but there are more than one. We don't support this at this time!");
+            }
```

Comment:
If only for debug should be wrapped in a conditional?

Reply:
Not debug right now, an actual limitation until #245

Reply:
A comment here would be nice then just to explain that it's future debug.

Alternatively set it up as a debug conditional and comment out the conditional with a TODO